### PR TITLE
Implement client support for fast / offline bootstrap RPC

### PR DESCRIPTION
**Pull request by chromakode**
This replaces our old bootstrap process with a single RPC with a smaller response. This RPC also returns useful results while offline, allowing us to load the chat UI from a cold start without a connection.

@keybase/react-hackers

### CODE REVIEW

File: shared/reducers/config.js [hunk 1]
```diff
@@ -100,32 +117,32 @@ export default function (state: ConfigState = initialState, action: Action): Con
...
       return state
 
-    case Constants.bootstrapAttemptFailed: {
+    case Constants.bootstrapLoaded:
+      const {bootstrapStatus} = action.payload
       return {
         ...state,
-        bootstrapTriesRemaining: state.bootstrapTriesRemaining - 1,
+        ...bootstrapStatus,
+        following: arrayToObjectSet(bootstrapStatus.following),
```
Comment (chromakode) [inline shared/reducers/config.js]: The service gives us to these as arrays, but we expect them to be objects w/ each key having value `true`. Would be nice to convert these to `Set`s in the future.

File: shared/actions/login/index.js [hunk 2]
```diff
@@ -34,47 +34,48 @@ const waitingForResponse = (waiting: boolean) : TypedAction<'login:waitingForRes
...
-    } else if (status.registered) { // relogging in
-      dispatch(getAccounts())
-      dispatch(navigateTo(['login'], [loginTab]))
-    } else { // no idea
-      dispatch(navigateTo([loginTab]))
+    } else {
+      dispatch(navigateTo([profileTab]))
     }
+  } else if (registered) { // relogging in
+    Promise.all([dispatch(getExtendedStatus()), dispatch(getAccounts())]).then(() => {
```
Comment (chromakode) [inline shared/actions/login/index.js]: This'll probably be rewritten to merge around @MarcoPolo's sagaization of this file.

**Comment by chrisnojima** [comment]
Note to myself: this should go in before https://github.com/keybase/client/pull/6628

File: shared/actions/login/index.js (hunk 2 above)
Comment (chrisnojima) [inline shared/actions/login/index.js]: yeah we should ditch all thunks that return promises i think

File: shared/reducers/config.js [hunk 3]
```diff
@@ -48,10 +51,24 @@ const initialState: ConfigState = {
...
+  registered: false,
   readyForBootstrap,
   status: null,
   uid: null,
   username: null,
+  deviceID: null,
+  deviceName: null,
+}
+
+function arrayToObjectSet (arr) {
```
Comment (chrisnojima) [inline shared/reducers/config.js]: could be something like

```
return (arr || []).reduce((obj, key) => {...obj, [key]: true}, {})
```

but its a 🐑

**Comment by patrickxb** [comment]
@chromakode if you rebase, the CI error should be fixed.

**Pull request merged**